src/scraper/scraper.cpp

```diff
@@ -5243,104 +5240,73 @@ scraperSBvalidationtype ValidateSuperblock(const NN::Superblock& NewFormatSuperb
                         }
                     }
                 }
+                if (fDebug3) _log(logattribute::INFO, "ENDLOCK", "CScraperManifest::cs_mapManifest");
+            } // Critical section scope for cs_mapManifest.
 
-                bool bMatched = false;
 
-                // Get an iterator pointing to the project in the project index map and match to the hint.
+            // Walk the time map (backwards in time because the sort order is descending), and select the first
+            // Project Part (Object) content hash that meets the convergence rule.
+            for (const auto& iter : mProjectObjectsBinnedByTime)
+            {
+                // Here we only consider those objects from manifests that have a time that is equal to or before the superblock to be validated,
+                // and which matches the project (part) hint.
+                uint32_t nReducedProjectObjectContentHash = std::get<0>(iter.second).Get64() >> 32;
+
                 if (const auto SBProjectIter = NewFormatSuperblock.m_projects.Try(iWhitelistProject.m_name))
                 {
                     // Pull the convergence hint (reduced content hash) for the project object.
                     uint32_t nReducedSBProjectObjectContentHash = SBProjectIter->m_convergence_hint;
 
-                    for (const auto& iter : mProjectObjectsBinnedbyContent)
+                    if (iter.first <= NewFormatSuperblock.m_timestamp && nReducedProjectObjectContentHash == nReducedSBProjectObjectContentHash)
```

Comment:
If a node's time is behind, this could result in a false validation failure. A scraper could publish a manifest with a time ahead of the perceived time that a node received the superblock. As in...

1. Scraper detects new etag, pulls the stats, and publishes a new manifest  
2. 10 seconds later, a node 20 seconds behind stakes a superblock with that manifest 
3. Another node immediately receives the manifest and the superblock...
    manifest time: _t_ - 10, 
    superblock time: _t_ - 20, 
    where _t_ is roughly the node's current time.
4. Validation fails because the manifest was not considered when reconstructing the convergence

The superblock timestamp will be set to the timestamp of its containing block when received by a node. I don't think we need to check the timestamp here because the content hash should adequately filter out ineligible candidates.

Reply:
Hmm... I put this in there to try and conform the validation to the existing implementation of the formation of a convergence from the source node, while trying to deal with the exact opposite case that you describe above, which is that the convergence that is hinted by the incoming superblock may not be the latest by project convergence that can be formed.

Reply:
We may have to check every combination of the matching project parts (breadth-and-most-recent-first) to also protect against temporary network partitions or delays during the superblock window. Not a pretty algorithm...

Reply:
Yep. And it is 99.9% likely the head of the list. But it is the 0.1% I am worried about. I do not want to get into an 2^n (n the number of projects) algorithm here for a 0.1% corner case. I have an idea. I will have to draw it out on a picture.

Reply:
![image](https://user-images.githubusercontent.com/7529186/66235418-02672680-e6be-11e9-97bf-1019bc0bdaec.png)
![image](https://user-images.githubusercontent.com/7529186/66235437-0dba5200-e6be-11e9-983b-5a75794e9d6f.png)
![image](https://user-images.githubusercontent.com/7529186/66235469-190d7d80-e6be-11e9-9fba-944364c2b38d.png)

Reply:
> _rare_

To say the least :slightly_smiling_face: 

And this only matters in a fallback-to-project-level convergence.

Reply:
When the node has only been running for a short enough while that it is not cached...

But we will do the final algo anyway, because I like to be thorough. I don't think we should delay the interim release over this. Let's merge this tracking PR as is and continue our work on this on integrated_scraper_2.